app/options.go

```diff
@@ -223,6 +232,7 @@ type options struct {
 	Voters                   int
 	StandBys                 int
 	RolesAdjustmentFrequency time.Duration
+	OnRolesAdjustment        func(*client.Client, []client.NodeInfo) error
```

Comment:
Only remaining question---would it make more sense for this to take a node ID instead of the client object, if the motivating use-case is to check whether the current leader is the same as the local node? I'm having trouble thinking of a good way to make use of the actual connection in this context.

Reply:
Yes, I suppose fetching the leader info directly and passing that through would make more sense.

Reply:
Updated now. The hook takes 2 arguments, a `client.NodeInfo` for the leader and a `[]client.NodeInfo` containing all nodes.